analyzers/src/SonarAnalyzer.Common/Rules/SymbolicExecutionRunnerBase.cs

```diff
@@ -81,6 +82,12 @@ protected void Analyze(SonarAnalysisContext sonarContext, SyntaxNodeAnalysisCont
             }
         }
 
+        protected static bool IsInLinqExpression(SyntaxNodeAnalysisContext context) =>
```

Comment:
That's too complicated. I've left FP comment to document the nested case.

Lambda nested inside expression doesn't seem to be supported anyway.

Reply:
What do you think about putting the check into `NullPointerDereferenceBase`: Before reporting, check if the node in question is in an expression tree and don't report if. 

We need this kind of check on multiple other occasions too, because in an expression context, some newer language features don't compile (`?.` for instance). We likely have rules where we tell users to use feature x instead of y, and we should not report if y isn't supported in an expression context. Here is a list of the features affected:

https://github.com/dotnet/csharplang/discussions/4727

Reply:
Doing it inside the engine is too late, we don't need to analyze those, as other rules will likely make no sense too. We can save a bit of performance

Reply:
I made a copy of those